ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -384,16 +403,169 @@ def all_mux_status(vm_set):
             all_mux_status[bridge] = get_mux_status(vm_set, port_index)
         return jsonify(all_mux_status)
     except Exception as e:
-        return jsonify({'err_msg': 'Get all mux status failed, vm_set: {}, exception: {}'.format(vm_set, repr(e))}), 500
+        err_msg = 'Get all mux status failed, vm_set: {}, exception: {}'.format(vm_set, repr(e))
+        app.logger.error('{} {} {}'.format(request.method, request.url, err_msg))
+        return jsonify({'err_msg': err_msg}), 500
+
+
+def _validate_out_ports(data):
+    """Validate the posted data for updating flow action.
+
+    Args:
+        data (dict): Posted json data. Expected:
+                {"out_ports": [<port>, <port>, ...]}
+            where <port> could be "nic", "tor_a" or "tor_b".
+
+    Returns:
+        tuple: Return the result in a tuple. The first item is either True or False. The second item is extra message.
+    """
+    supported_out_ports = ['nic', 'tor_a', 'tor_b']
+    try:
+        assert 'out_ports' in data, 'Missing "out_ports" field'
+        for port in data['out_ports']:
+            assert port in supported_out_ports, 'Unsupported port: "{}", supported: {}'.format(port, supported_out_ports)
+        return True, ''
+    except Exception as e:
+        return False, 'Validate out_ports {} failed with exception: {}'.format(json.dumps(data), repr(e))
+
+
+def update_flow_action_nic(mux_status, action):
```

Comment:
It's better to change the method name to ```update_flow_action_to_nic``` for better readability.

Reply:
Sure, updated.